From 31f022cc8f78692274782706edc2d7a82294c1fb Mon Sep 17 00:00:00 2001
From: Manish Dharanenthiran <quic_mdharane@quicinc.com>
Date: Thu, 5 Dec 2024 16:41:00 +0530
Subject: [PATCH] UPSTREAM: Protocol: wifi: mac80211: Add NULL check before
 accessing scan request

During offchannel TX, ieee80211_is_radar_required is called with scan
request argument to check the hw index field. But, there can be cases
where the local->scan_req can be NULL. Accessing this field can lead
NULL pointer access.

Avoid this by returning false by default if the scan request field is
not set.

Patch-Dependency: 716-ath12k-AP-bringup-with-5G-Low-5G-High-in-dfs-channel.patch

Signed-off-by: Manish Dharanenthiran <quic_mdharane@quicinc.com>
---
 net/mac80211/chan.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/net/mac80211/chan.c b/net/mac80211/chan.c
index 81fd5bc..d87c123 100644
--- a/net/mac80211/chan.c
+++ b/net/mac80211/chan.c
@@ -615,6 +615,11 @@ bool ieee80211_is_radar_required(struct ieee80211_local *local,
 			if (link && link->radar_required) {
 				struct ieee80211_channel *chan = link->conf->chandef.chan;
 
+				if (!req) {
+					rcu_read_unlock();
+					return false;
+				}
+
 				hw_idx = cfg80211_get_hw_idx_by_chan(wiphy, chan);
 				if (hw_idx == req->hw_idx) {
 					rcu_read_unlock();
-- 
2.34.1

